### DISCUSSION TITLE
Update()에서 For 반복문 한번만 실행 어떻게 해요?

### DISCUSSION THREAD

**Discussion by janggf** [naverz/zepeto-studio-kor#524] [5. Q&A]
![캡처](https://user-images.githubusercontent.com/13530263/159662564-803e6b57-8156-4b5b-bc74-e2c636d93129.JPG)

*상황 설명*
스크립트를 부착한 부모 밑에는  bars들이 있습니다 작동은 this.bars[0]. 이거만 되고요
한번만 깔끔하게 찾고 색깔 바꾸고 했으면 하네요

아래는 해결가능한 다른방법
this.bars[0].
this.bars[1].
this.bars[2].
이건 너무 노가다에요

**Comment by minjujuu** [comment]
한 번만 해야 하는 거면 Start() 에서 해보시면 될 것 같습니다

**Reply by janggf** [reply]
그건 안되요 특정 상황에 변해야합니다

**Reply by minjujuu** [reply]
그럼 이런식으로 bool 값을 통해 제어하시는 건 어떨까요?
```
isCompleted:boolean = false;
FixedUpdate() {
      if(isCompleted == false) {
            isCompleted = true;
            for문 { }
      }
}
```

**Reply by Zepeto-TechEvangelist** [reply]
flag 변수를 선언해서 조건문 안에 for 문을 넣어주세요.
for문이 돌고나면 flag 값을 바꿔주면 딱 한번만 실행될거에요.
